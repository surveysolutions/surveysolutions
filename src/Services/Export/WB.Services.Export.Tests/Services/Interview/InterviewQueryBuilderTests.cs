using ApprovalTests;
using ApprovalTests.Core;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using NUnit.Framework;
using WB.Services.Export.Interview;
using WB.Services.Export.Questionnaire;

namespace WB.Services.Export.Tests.Services.Interview
{
    [TestOf(typeof(InterviewQueryBuilder))]
    [UseApprovalSubdirectory("approved")]
    [IgnoreLineEndings(true)]
    [UseReporter(typeof(DiffReporter), typeof(NUnitReporter))]
    public class InterviewQueryBuilderTests
    {
        [Test]
        public void should_build_query_for_group_with_question()
        {
            var questionnaire = Create.QuestionnaireDocument(Id.gA, 6, "quest", Create.Group(Id.gB, "group1",
                Create.NumericIntegerQuestion(variable: "num1")));
            questionnaire.ConnectChildrenWithParent();

            // Act
            var query = InterviewQueryBuilder.GetInterviewsQuery(questionnaire.Find<Group>(Id.gB));

            // Assert
            Approvals.Verify(query);
        }

        [Test]
        public void should_be_able_to_build_query_for_roster()
        {
            var questionnaire = Create.QuestionnaireDocument(Id.gA, 6, "quest", 
                Create.Roster(Id.gB, variable: "roster1",
                    children: new IQuestionnaireEntity[] { Create.NumericIntegerQuestion(variable: "num1") }));
            questionnaire.ConnectChildrenWithParent();

            // Act
            var query = InterviewQueryBuilder.GetInterviewsQuery(questionnaire.Find<Group>(Id.gB));

            // Assert
            Approvals.Verify(query);
        }
    }
}
