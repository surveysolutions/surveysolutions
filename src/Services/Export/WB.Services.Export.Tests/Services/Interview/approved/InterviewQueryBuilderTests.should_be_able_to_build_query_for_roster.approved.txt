select data.interview_id as data_interview_id, data.roster_vector as data_roster_vector,  data."num1" as "data_num1" ,  enablement."num1" as "enablement_num1" ,  validity."num1" as "validity_num1"  from 
"tenant"."qqqqqqqqqqqqqqqqqqqqqg$6_roster1" data 
    INNER JOIN "tenant"."qqqqqqqqqqqqqqqqqqqqqg$6_roster1_e" enablement ON data.interview_id = enablement.interview_id
    INNER JOIN "tenant"."qqqqqqqqqqqqqqqqqqqqqg$6_roster1_v" validity ON data.interview_id = validity.interview_id
 WHERE data.interview_id = ANY(@ids)